portable/GCC/ARM_CRx_MPU/portmacro.h

```diff
@@ -0,0 +1,690 @@
+ * is changed to use this region when a call to a function is made that goes through
+ * the MPU Wrappers.
+ *
+ * This value should be a size that can be made into a MPU Region.
+ *
+ */
+#ifndef configSYSTEM_CALL_STACK_SIZE
+    #error configSYSTEM_CALL_STACK_SIZE must be defined to the desired size of the system call stack in words for using MPU wrappers v2.
+#endif /* configSYSTEM_CALL_STACK_SIZE */
+
+#if( configUSE_PORT_OPTIMISED_TASK_SELECTION == 1 )
+    /* Check the configuration. */
+    #if( configMAX_PRIORITIES > 32 )
+        #error "configUSE_PORT_OPTIMISED_TASK_SELECTION can only be set to 1 when " \
+                "configMAX_PRIORITIES is less than or equal to 32." \
+                "It is very rare that a system requires more than 10 to 15 difference " \
+                "priorities as tasks that share a priority will time slice."
+    #endif /* ( configMAX_PRIORITIES > 32 ) */
+
+    /** @brief Mark that a task of the current priority is ready for execution
+     *
+     * @ingroup Scheduler
+     *
+     * @param[in] uxPriority Priority of task that can enter the ready state
+     * @param[out] uxTopReadyPriority Bitmap of tasks that are in the ready state
+     */
+    #define portRECORD_READY_PRIORITY( uxPriority, uxTopReadyPriority ) \
+        ( uxTopReadyPriority ) |= ( 1UL << ( uxPriority ) )
+
+    /** @brief Mark that a task of the current priority has left the ready state
+     *
+     * @ingroup Scheduler
+     *
+     * @param[in] uxPriority Priority of task that is leaving the ready state
+     * @param[out] uxTopReadyPriority Bitmap of tasks that are in the ready state
+     */
+    #define portRESET_READY_PRIORITY( uxPriority, uxTopReadyPriority ) \
+        ( uxTopReadyPriority ) &= ~( 1UL << ( uxPriority ) )
+
+    /** @brief Determine what the highest priority ready task is.
+     *
+     * @ingroup Scheduler
+     *
+     * @param[out] uxTopReadyPriority Bitmap of tasks that are in the ready state
+     * @param[out] uxTopPriority The highest priority ready task's priority value.
+     */
+    #define portGET_HIGHEST_PRIORITY( uxTopPriority, uxTopReadyPriority ) \
+        uxTopPriority = ( 31UL - ulPortCountLeadingZeros( ( uxTopReadyPriority ) ) )
+
+#endif /* configUSE_PORT_OPTIMISED_TASK_SELECTION */
+
+#ifndef configSETUP_TICK_INTERRUPT
+    #error configSETUP_TICK_INTERRUPT() must be defined in FreeRTOSConfig.h to call the function that sets up the tick interrupt.
+#endif /* configSETUP_TICK_INTERRUPT */
+
+#ifndef configCLEAR_TICK_INTERRUPT
+    #error configCLEAR_TICK_INTERRUPT must be defined in FreeRTOSConfig.h to clear which ever interrupt was used to generate the tick interrupt.
+#endif /* configCLEAR_TICK_INTERRUPT */
+
+#ifndef configEOI_ADDRESS
+    #error Did not define a memrory address to mark the end of an Interrupt
+#endif /* configEOI_ADDRESS */
+
+#ifdef configUSE_TICKLESS_IDLE
+    #if( configUSE_TICKLESS_IDLE != 0 )
+        #error This port does not yet support tickless idle
```

Comment:
Minor nitpick, i would remove the word 'yet'

Reply:
```suggestion
        #error This port does not support tickless idle
```